### PR TITLE
ucommon 6.3.0

### PR DESCRIPTION
**Pull request by bfontaine**

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Install is failing, possibly CXX11 stuff: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/20559/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_ucommon/

**Comment by DomT4** [comment]
Build is failing with everything but Mountain Lion. Try something like ` needs :cxx11 ` and ` ENV.cxx11 `. That tends to be ML.

**Comment by DomT4** [comment]
Whoops. Sorry, Mike and I commented at the same time.

**Comment by bfontaine** [comment]
Thanks, I was googling the error, I’m not familiar with C++.

**Comment by DomT4** [comment]
I'd presume ` needs :cxx11 ` gets ignored unless it's called with an ` ENV `. Not sure though.

**Comment by bfontaine** [comment]
Yeah it failed without ENV.cxx11. Just added it, we’ll see.

**Comment by bfontaine** [comment]
Nope, still a failure (on ML too now).

**Comment by DomT4** [comment]
Try a stricter ` fails_with :clang `.

**Comment by bfontaine** [comment]
You’re the best, it works! Any idea about what I can add as a reason?

**Comment by DomT4** [comment]
I'd guess it's accidentally pulling in a wrong flag or something. GNU stuff can tend to struggle with setting the right environment for Clang at times. You can always go for something as basic as ` cause "Clang causes fatal compile failures on Mountain Lion" ` or similar.

**Comment by bfontaine** [comment]
I made [a diff between the two versions](https://gist.github.com/bfontaine/9d2d78d9434928825d07), *if we’re lucky* I could find where’s the problem and make a patch.

**Comment by DomT4** [comment]
[This aspect](https://gist.github.com/bfontaine/9d2d78d9434928825d07#file-ucommon-6-2-2-to-6-3-0-diff-L1791-L1805) looks interesting. Looks like they changed a default around stdlib. @mistydemeo helped me out on ML compiler issues before now, she may be able to offer some advice here.

**Comment by bfontaine** [comment]
I tried with patch which revert these changes but the build error stays the same (on my Mac, OS X 10.10).

**Comment by DomT4** [comment]
IMO, If ` fails_with :clang ` worked before, stick that back in and submit the clang failure to GNU upstream and add a note in the formula as to where the bug submission is located and note the clang failure should be reviewed in a future update.

File: Library/Formula/ucommon.rb [hunk 1]
```diff
@@ -14,16 +12,24 @@ class Ucommon < Formula
   depends_on "pkg-config" => :build
   depends_on "gnutls"
 
+  needs :cxx11
+
+  fails_with :clang
```
Comment (mikemcquaid) [inline Library/Formula/ucommon.rb]: We should have a `build` and `cause` here too if this works.

Comment (bfontaine) [inline Library/Formula/ucommon.rb]: @mikemcquaid Yes, that was just a test, I need to edit the formula with a comment pointing to the bug report.

**Comment by bfontaine** [comment]
@DomT4 I just sent a mail on the project mailing-list, I’m waiting for it to appear in the [archives](https://lists.gnu.org/archive/html/bug-gnucomm/) and I’ll add a link in a comment

File: Library/Formula/ucommon.rb (hunk 1 above)
Comment (bfontaine) [inline Library/Formula/ucommon.rb]: @mikemcquaid What can I add as a cause except “it doesn’t build”? cc @DomT4

Comment (mikemcquaid) [inline Library/Formula/ucommon.rb]: The error produced by the compiler.

**Comment by bfontaine** [comment]
@mikemcquaid @DomT4 I added a comment as well as the requested `build` and `cause`. Let me know if I need to do something else.

File: Library/Formula/ucommon.rb [hunk 2]
```diff
@@ -14,16 +12,29 @@ class Ucommon < Formula
   depends_on "pkg-config" => :build
   depends_on "gnutls"
 
+  needs :cxx11
+
+  # Upstream bug report:
+  #   http://lists.gnu.org/archive/html/bug-commoncpp/2015-01/msg00000.html
```
Comment (DomT4) [inline Library/Formula/ucommon.rb]: Indentation is slightly off here. Otherwise, this looks alright to me.

Comment (bfontaine) [inline Library/Formula/ucommon.rb]: It’s intentional, to show that this line is linked to the previous one :smiley: 

e.g.:

```
# something:
#   foo
#   bar
# something else:
#   qux
```

vs:
```
# something:
# foo
# bar
# something else:
# qux
```

Comment (DomT4) [inline Library/Formula/ucommon.rb]: Oh, I usually just keep them indented on the same level. I guess it's entirely stylistic preference though.

Comment (bfontaine) [inline Library/Formula/ucommon.rb]: Yeah, for me it means “the previous line continues here but I needed to break it to keep the text in ≤80 columns”.

**Comment by mikemcquaid** [comment]
Just to check: has the Clang issue been reported upstream to GNU? My concern is that if this doesn't work in Clang on any of our OSs then perhaps it's something we should hold off on merging. I'm sorry about that :(

**Comment by bfontaine** [comment]
@mikemcquaid Yep, [reported here](http://lists.gnu.org/archive/html/bug-commoncpp/2015-01/msg00000.html).

**Comment by mikemcquaid** [comment]
Oh, thanks, sorry I missed that.

**Comment by bfontaine** [comment]
No response from upstream, I’m closing this one.

**Pull request closed without merging**